components/zoom_admin/zoom_admin.app.mjs

```diff
@@ -0,0 +1,374 @@
+      opts.headers["user-agent"] = "@PipedreamHQ/pipedream v0.1";
+      const { path } = opts;
+      delete opts.path;
+      // eslint-disable-next-line multiline-ternary
+      opts.url = `${this._apiUrl()}${path[0] === "/" ? "" : "/"}${path}`;
+      return await axios(opts);
+    },
+    sanitizedArray(value) {
+      if (isArray(value)) {
+        return value.map((item) => get(item, "value", item));
+      }
+
+      // If is string, try to convert it in an array
+      if (isString(value)) {
+        // Return an empty array if string is empty
+        if (isEmpty(value)) {
+          return [];
+        }
+
+        return value.replace(/["'[\]\s]+/g, "").split(",");
+      }
+
+      throw new Error(`${value} is not an array or an array-like`);
+    },
+    async listMeetings(nextPageToken) {
+      const { data } = await this._makeRequest({
+        path: "/users/me/meetings",
+        params: {
+          page_size: 100,
+          next_page_token: nextPageToken,
+        },
+      });
+
+      return data;
+    },
+    async listMeetingsOccurrences(meetingId, isWebinar) {
+      try {
+        const path = isWebinar
+          ? `/webinars/${meetingId}`
+          : `/meetings/${meetingId}`;
+        const res = await this._makeRequest({
+          path,
+        });
+        return get(res, "data.occurrences", []);
+      } catch {
+        return [];
+      }
+    },
+    async listWebinars({
+      pageSize,
+      nextPageToken,
+    }) {
+      const { data } = await this._makeRequest({
+        path: "/users/me/webinars",
+        params: {
+          page_size: pageSize || 300,
+          next_page_token: nextPageToken,
+        },
+      });
+      return data;
+    },
+    async listWebinarPanelists(webinarID) {
+      const { data } = await this._makeRequest({
+        path: `/webinars/${webinarID}/panelists`,
+      });
+      return data;
+    },
+    async listMeetingRegistrants(meetingId, occurrenceId, pageNumber, status, isWebinar) {
+      const path = isWebinar
+        ? `/webinars/${meetingId}/registrants`
+        : `/meetings/${meetingId}/registrants`;
+      try {
+        const res = await this._makeRequest({
+          path,
+          params: {
+            occurrence_id: occurrenceId,
+            page_size: 30,
+            page_number: pageNumber,
+            status,
+          },
+        });
+
+        if (pageNumber > get(res, ("data.page_count"))) {
+          return [];
+        }
+
+        return get(res, "data.registrants", []);
+      } catch (err) {
+        console.error(err);
+        return [];
+      }
+    },
+    async listUserRecordings(nextPageToken, pageNumber) {
+      const res = await this._makeRequest({
+        path: "/users/me/recordings",
+        params: {
+          page_size: 30,
+          next_page_token: nextPageToken,
+        },
+      });
+
+      if (pageNumber > get(res, ("data.page_count"))) {
+        return [];
+      }
+
+      return get(res, "data");
```

Comment:
I my opinion I would say is not ideal to deal with data logic in these functions. For me they just need to do one job and that is to return the response right from the api call using in this case `axios` lib.